akka-stream-tests/src/test/scala/akka/stream/scaladsl/AttributesSpec.scala

```diff
@@ -25,6 +25,49 @@ object AttributesSpec {
     }
   }
 
+  class AttributesFlow(_initialAttributes: Attributes = Attributes.none) extends GraphStageWithMaterializedValue[FlowShape[Any, Any], Attributes] {
+
+    val in = Inlet[Any]("in")
+    val out = Outlet[Any]("out")
+
+    override protected def initialAttributes: Attributes = _initialAttributes
+    override val shape = FlowShape(in, out)
+    override def createLogicAndMaterializedValue(inheritedAttributes: Attributes): (GraphStageLogic, Attributes) = {
+      val logic = new GraphStageLogic(shape) {
+
+        setHandlers(in, out, new InHandler with OutHandler {
+          override def onPush(): Unit = push(out, grab(in))
+          override def onPull(): Unit = pull(in)
+        })
+      }
+
+      (logic, inheritedAttributes)
+    }
+  }
```

Comment:
yeah good one for testing those things (made the same, though pushing the attrs lol :))